hydroflow_plus_test/src/cluster/two_pc.rs

```diff
@@ -0,0 +1,121 @@
+    let c_participant_voted_abort = c_received_reply
+        .clone()
+        .filter(q!(|(_id, (_t, reply))| reply == "abort"))
+        .map(q!(|(id, (t, _reply))| (t, id)));
+    let p_receive_abort = c_participant_voted_abort.broadcast_bincode(&participants);
+    p_receive_abort.clone().inspect(q!(|(t, id)| println!(
+        "{} vote abort for transaction {}",
+        id, t
+    )));
+    let c_receive_ack = p_receive_abort.send_bincode(&coordinator);
+    c_receive_ack.for_each(q!(|(id, (t, _))| println!(
+        "Coordinator receive participant {} abort for transaction {}",
+        id, t
+    )));
+
+    // committed transactions
+    let c_participant_voted_commit = c_received_reply
+    .filter(q!(|(_id, (_t, reply))| reply == "commit"))
+    .map(q!(|(id, (t, _reply))| (t, id)))
+    // fold_keyed: 1 input stream of type (K, V1), 1 output stream of type (K, V2). 
+    // The output will have one tuple for each distinct K, with an accumulated value of type V2.
+    .tick_batch().fold_keyed(q!(|| 0), q!(|old: &mut u32, _: u32| *old += 1)).filter_map(q!(move |(t, count)| {
+        // here I set the participant to 3. If want more or less participant, fix line 26 of examples/broadcast.rs
+        if count == num_participants {
+            Some(t)
+        } else {
+            None
+        }
+    }));
+
+    // broadcast commit transactions to participants.
+    let p_receive_commit = c_participant_voted_commit
+        .all_ticks()
+        .broadcast_bincode(&participants);
+    // p_receive_commit.clone().for_each(q!(|t| println!("commit for transaction {}", t)));
+
+    let c_receive_ack = p_receive_commit.send_bincode(&coordinator);
+    c_receive_ack.for_each(q!(|(id, t)| println!(
+        "receive participant {} commit for transaction {}",
+        id, t
+    )));
+    (coordinator, participants, client)
+}
+
+#[cfg(test)]
+mod tests {
+    use hydro_deploy::Deployment;
+    use hydroflow_plus::deploy::TrybuildHost;
+
+    #[tokio::test]
+    async fn two_pc() {
+        let mut deployment = Deployment::new();
+
+        let builder: hydroflow_plus::FlowBuilder<'_> = hydroflow_plus::FlowBuilder::new();
+        let num_participants: u32 = 3;
+        let (coordinator, participants, client) = super::two_pc(&builder, num_participants);
+        let built = builder.with_default_optimize();
+        let nodes = built
+            .with_process(&coordinator, TrybuildHost::new(deployment.Localhost()))
+            .with_cluster(
+                &participants,
+                (0..num_participants)
+                    .map(|_| TrybuildHost::new(deployment.Localhost()))
+                    .collect::<Vec<_>>(),
+            )
+            .with_process(&client, TrybuildHost::new(deployment.Localhost()))
+            .deploy(&mut deployment);
+        // println!("{:?}", built.ir());
+        deployment.deploy().await.unwrap();
+
+        deployment.start().await.unwrap();
+
+        tokio::signal::ctrl_c().await.unwrap();
```

Comment:
This unit test will hang because `Ctrl-C` won't be invoked; we should move this to an example for now.